src/lib.rs

```diff
@@ -94,6 +94,26 @@ impl<T> Node<T> {
             value,
         }
     }
+
+    fn map_value<U>(self, transform: impl FnOnce(T) -> U) -> Node<U> {
+        Node {
+            parent: self.parent,
+            prev_sibling: self.prev_sibling,
+            next_sibling: self.next_sibling,
+            children: self.children,
+            value: transform(self.value),
+        }
+    }
+
+    fn map_value_ref<U>(&self, transform: impl FnOnce(&T) -> U) -> Node<U> {
```

Comment:
I wonder if we should rather go the `Option`-like route and have a method like

```rust
fn as_ref(&self) -> Node<&T>;
```

here which can then be combined with the existing `map_value`.

Reply:
Yes, that's much better, indeed!

Reply:
I am not sure, but this might actually want to return `NodeRef<'_, T>` which then needs a map as well for everything to fit together? Would need to try it out...

Reply:
As discussed below, this is not public API and hence it is probably better to keep things aligned with the public API even if it is less general internally.